benchmarks/segmentation_layers.py

```diff
@@ -63,38 +264,145 @@ def _batch_norm_tensor():
 def batch_norm_nt_tensors_same_size():
     inputs = [torch.randn(C, H, W) for _ in range(N)]
     nt = nestedtensor.nested_tensor(inputs)
-    batch_norm = torch.nn.BatchNorm2d(30, 1e-05, 0.1)
     batch_norm.eval()
 
     def _batch_norm_nt():
         batch_norm(nt)
     
     return _batch_norm_nt
 
+def batch_norm_tensor_small_diff_pad():
+    inputs = get_input_small_diff(pad=True)
+    stack = torch.stack(inputs)
+    batch_norm.eval()
+
+    def _batch_norm_tensor_small_diff_pad():
+        batch_norm(stack)
+    return _batch_norm_tensor_small_diff_pad
+
+def batch_norm_tensor_small_diff_iter():
+    inputs = get_input_small_diff()
+    batch_norm.eval()
+
+    def _batch_norm_tensor_small_diff_iter():
+        for t in inputs:
+            batch_norm(t.unsqueeze(0)).squeeze(0)
+    return _batch_norm_tensor_small_diff_iter
+
+def batch_norm_nt_small_diff():
+    inputs = get_input_small_diff()
+    nt = nestedtensor.nested_tensor(inputs)
+    batch_norm.eval()
+
+    def _batch_norm_nt_small_diff():
+        batch_norm(nt)
+
+    return _batch_norm_nt_small_diff
+
+def batch_norm_tensor_big_diff_pad():
+    inputs = get_input_big_diff(pad=True)
+    stack = torch.stack(inputs)
+    batch_norm.eval()
+
+    def _batch_norm_tensor_big_diff_pad():
+        batch_norm(stack)
+    return _batch_norm_tensor_big_diff_pad
+
+def batch_norm_tensor_big_diff_iter():
+    inputs = get_input_big_diff()
+    batch_norm.eval()
+
+    def _batch_norm_tensor_big_diff_iter():
+        for t in inputs:
+            batch_norm(t.unsqueeze(0)).squeeze(0)
+    return _batch_norm_tensor_big_diff_iter
+
+def batch_norm_nt_big_diff():
+    inputs = get_input_big_diff()
+    nt = nestedtensor.nested_tensor(inputs)
+    batch_norm.eval()
+
+    def _batch_norm_nt_big_diff():
+        batch_norm(nt)
+
+    return _batch_norm_nt_big_diff
+
 #
 # max_pool2d
 #
+max_pool2d = torch.nn.MaxPool2d(kernel_size=(2, 2), stride=(2, 2), padding=(0, 0), dilation=(1, 1))
+
 def max_pool2d_tensor_tensors_same_size():
     inputs = [torch.randn(C, H, W) for _ in range(N)]
     a = torch.stack(inputs)
-    maxPool2d = torch.nn.MaxPool2d(30)
 
     def _max_pool2d_tensor():
-        maxPool2d(a)
+        max_pool2d(a)
     return _max_pool2d_tensor
 
 def max_pool2d_nt_tensors_same_size():
     inputs = [torch.randn(C, H, W) for _ in range(N)]
-    maxPool2d = torch.nn.MaxPool2d(30)
     nt = nestedtensor.nested_tensor(inputs)
+
     def _max_pool2d_nt():
-        maxPool2d(nt)
+        max_pool2d(nt)
 
     return _max_pool2d_nt
 
+def max_pool2d_tensor_small_diff_pad():
+    inputs = get_input_small_diff(pad=True)
```

Comment:
You could make this a parameter to a factory function that returns benchmark functions.

So like:

```
def gen_max_pool2d_tensor(gen_fn):
    def local_fn():
       inputs = gen_fn(pad=True)
       stack = torch.stack(inputs)

       def _max_pool2d_tensor_small_diff_pad():
           max_pool2d(stack)
       return _max_pool2d_tensor_small_diff_pad
   return local_fn

max_pool2d_tensor_small_diff_pad = gen_max_pool2d_tensor(get_input_small_diff)
```

Reply:
I think that can be applied to a multiple parts in this file.